src/platform_impl/web/stdweb/gamepad.rs

```diff
@@ -0,0 +1,77 @@
+use std::{cmp::PartialEq};
+use crate::platform_impl::platform::device;
+use super::utils;
+
+pub struct Gamepad {
+    pub(crate) index: i32,
+    pub(crate) raw: stdweb::web::Gamepad,
+    pub(crate) mapping: device::gamepad::Mapping,
+}
+
+impl Gamepad {
+    pub fn new(raw: stdweb::web::Gamepad) -> Self {
+        let mapping = utils::create_mapping(&raw);
+
+        Self {
+            index: raw.index(),
+            raw,
+            mapping,
+        }
+    }
+
+    pub fn raw(&self) -> stdweb::web::Gamepad {
+        self.raw.clone()
+    }
+
+    // An integer that is auto-incremented to be unique for each device
+    // currently connected to the system.
+    // https://developer.mozilla.org/en-US/docs/Web/API/Gamepad/index
+    pub fn index(&self) -> i32 {
+        self.raw.index()
+    }
+
+    // A string containing some information about the controller.
+    // https://developer.mozilla.org/en-US/docs/Web/API/Gamepad/id
+    pub fn id(&self) -> String {
+        self.raw.id()
+    }
+
+    // A boolean indicating whether the gamepad is still connected to the system.
+    // https://developer.mozilla.org/en-US/docs/Web/API/Gamepad/connected
+    pub fn connected(&self) -> bool {
+        self.raw.connected()
+    }
+
+    // EXPERIMENTAL
+    #[allow(dead_code)]
+    pub fn vibrate(&self, _value: f64, _duration: f64) {
+        //     for actuator in self.raw.haptic_actuators().values() {
+        //         actuator
+        //         .ok()
+        //         .and_then(|a| match a.type_ {
+        //             web_sys::GamepadHapticActuatorType::Vibration => {
+        //                 a.pulse(value, duration);
+        //                 Some(())
+        //             },
+        //             _ => None,
+        //         });
+        //     }
```

Comment:
It's fine to leave this as a TODO, just make sure it gets noted in the public API at the top level.